### PR TITLE
PLANET-7446: Replace lightbox functionality

### PR DESCRIPTION
**Pull request by mardelnet**
Ref:
- https://jira.greenpeace.org/browse/PLANET-7446
- https://jira.greenpeace.org/browse/PLANET-7510

<hr>

**DESCRIPTION:**
This PR is a proof of concept that attempts to remove the custom lightbox functionality and use the native lightbox as a replacement. This PR implements the native lightbox feature in the core image block and the Planet4 gallery block. When expanded, the native lightbox shows the image caption, however, I could not replicate the navigation between images that the custom lightbox offers.

<hr>

**TESTING:**
- https://www-dev.greenpeace.org/test-janus/planet-7446-replace-lightbox-functionality/
- Note: disabling the P4 plugin is recommended, as it contains logic of the gallery block and the custom lightbox.

### CODE REVIEW

**Comment by planet-4** [comment]
### Test instance is ready :rocket:

:new_moon: [janus](https://www-dev.greenpeace.org/test-janus) | [admin](https://www-dev.greenpeace.org/test-janus/wp-admin/) | [blocks report](https://www-dev.greenpeace.org/test-janus/wp-admin/?page=plugin_blocks_report) | [CircleCI](https://app.circleci.com/pipelines/github/greenpeace/planet4-test-janus) | [composer-local.json](https://github.com/greenpeace/planet4-test-janus/blob/main/composer-local.json)

:watch: 2024.10.29 13:58:26

**Comment by mleray** [comment]
@mardelnet the caption is not working properly in the lightbox for the Image block, I either see no caption or the one previously shown in the Gallery block. Also, right now this implementation doesn't work for the Media & Text it seems?

File: assets/src/blocks/Gallery/GalleryBlock.js [hunk 1]
```diff
@@ -35,6 +35,10 @@ const attributes = {
     type: 'array',
     default: [],
   },
+  expand_on_click: {
+    type: 'boolean',
+    default: false,
```
Comment (mleray) [inline assets/src/blocks/Gallery/GalleryBlock.js:40]: Wondering if it should be true by default 🤔

Comment (mleray) [reply assets/src/blocks/Gallery/GalleryBlock.js:40]: In this case we might need to update existing blocks with a deprecated file

File: assets/src/blocks/Gallery/GalleryEditor.js [hunk 2]
```diff
@@ -75,6 +75,23 @@ const renderEdit = (attributes, setAttributes, isSelected) => {
...
+                {__('Expand on click', 'planet4-blocks-backend')}
+              </strong>
+              <p className="components-base-control__help">
+                {__('Scales the gallery images with a lightbox effect.', 'planet4-blocks-backend')}
+              </p>
+              <ToggleControl
+                label={__('Expand', 'planet4-blocks-backend')}
+                value={expand_on_click}
+                checked={expand_on_click}
+                help={expand_on_click ? __('Expand is enabled.', 'planet4-blocks-backend') : __('Expand is disabled.', 'planet4-blocks-backend')}
```
Comment (mleray) [inline assets/src/blocks/Gallery/GalleryEditor.js:90]: I'm not sure this helps very much 😁

File: assets/src/blocks/Gallery/GalleryFrontend.js [hunk 3]
```diff
@@ -55,16 +37,11 @@ export const GalleryFrontend = ({
 
       {images.length ? (
         <>
-          {layout === 'slider' && <GalleryCarousel onImageClick={openLightbox} images={images} />}
-          {layout === 'three-columns' && <GalleryThreeColumns onImageClick={openLightbox} images={images} postType={postType} />}
-          {layout === 'grid' && <GalleryGrid onImageClick={openLightbox} images={images} />}
+          {layout === 'slider' && <GalleryCarousel images={images} expand={expand} />}
```
Comment (mleray) [inline assets/src/blocks/Gallery/GalleryFrontend.js:40]: Maybe we could create a `galleryBlock` constant `{images, expand}` and use this for all the styles? Would remove some duplications

File: assets/src/blocks/Gallery/GalleryThreeColumns.js [hunk 4]
```diff
@@ -1,26 +1,28 @@
...
-              style={{objectPosition: image.focus_image}}
-              alt={image.alt_text}
-              title={image.alt_text}
-              className={`img_${postType}`}
-              data-index={index}
-              onClick={onImageClick}
-              role="presentation"
-            />
+          {
+            expand ?
```
Comment (mleray) [inline assets/src/blocks/Gallery/GalleryThreeColumns.js:13]: This code piece is repeated 3 times (one per style), maybe it could be refactored?

**Comment by mardelnet** [comment]
> This also definitely needs the design team's input, can you ask for UAT @mardelnet?

Absolutely @mleray!
However, considering this PR is a proof of concept, I'd like to know @comzeradd opinion on whether it makes sense to keep working on this implementation. I think that, besides the navigation between images when the lightbox is expanded, the rest of the features of our custom lightbox can be implemented in the native one.

**Comment by mardelnet** [comment]
I am closing this PR since the design team decided not to proceed with the native lightbox because it doesn't implement navigation between slides.

**Pull request closed without merging**